src/audio/pipewire.c

```diff
@@ -334,6 +333,8 @@ static int pipewire_stop(AudioID *id)
     message(3, "stopping audio playback");
     // unset the running flag
     state->is_running = false;
+    // drain the stream, to make sure there are no more samples playing, to avoid crackling
+    pw_stream_flush(state->stream, false);
```

Comment:
I'm thinking this should probably be done after the pw_stream_set_active call below? otherwise the thread might have time to push some samples after the flush, and thus bringing some crackling (though more rarely). This stop code probably deserves proof-reading from a pipewire guy.

Reply:
apparently, stop calls flush on its own, so I will remove that soon. The pipewire person said it's ok how I did it, so yeah, dk what could be bringing on the cracling

Reply:
Which stop?

Reply:
when I set the state of the stream to inactive, that's like a stop operation, which apparently flushes the stream too